crates/hir-ty/src/infer.rs

```diff
@@ -118,9 +118,12 @@ pub(crate) fn infer_query(db: &dyn HirDatabase, def: DefWithBodyId) -> Arc<Infer
 /// This is appropriate to use only after type-check: it assumes
 /// that normalization will succeed, for example.
 pub(crate) fn normalize(db: &dyn HirDatabase, trait_env: Arc<TraitEnvironment>, ty: Ty) -> Ty {
-    if !ty.data(Interner).flags.intersects(TypeFlags::HAS_PROJECTION) {
```

Comment:
I would believe this to ruin performance, could you check that? Because if it does I don't think we want to merge this.

Reply:
I changed that, and per the CI time we are good again. But it is not an important part of the PR, we can remove that (at cost of not being able to find layout for `struct Something { some_field: [Ty; SOME_CONST] }`, the original issue works without this)